mitmproxy/protocol/http.py

```diff
@@ -198,7 +205,7 @@ def __call__(self):
                 # It may be useful to pass additional args (such as the upgrade header)
                 # to next_layer in the future
                 if flow.response.status_code == 101:
-                    layer = self.ctx.next_layer(self)
+                    layer = self.ctx.next_layer(self, flow)
```

Comment:
Maybe - the code was already there - I figured the person who placed it here had a reason... 😄 
However, I think we should keep it like that.
We can use this mechanism to add h2c support (HTTP/2 over plain TCP without TLS) in the future.